### ISSUE TITLE
Error in the generated YAML

### ISSUE THREAD

**Issue by 372046933** [kubeflow/kubeflow#1524]
The following error pops after run `ks apply -f YAML` where `YAML` is generate by `ks show default`
```
error validating data: ValidationError(CustomResourceDefinition.spec): missing required field "scope" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1beta1.CustomResourceDefinitionSpec; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by kunmingg** [comment]
Can you provide which version are you using? 
And cmds you executed?

**Comment by kkasravi** [comment]
@372046933 might be referring to tf-job-operator.libsonnet 
Both CRD's do not include scope 
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/tf-job-operator.libsonnet#L36
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/tf-job-operator.libsonnet#L53

I've included scope in PR #1535 but the PR is dependent on [upgrading to ksonnet 0.12.0](https://github.com/kubeflow/kubeflow/pull/1508)

**Comment by 372046933** [comment]
Thank you. I was using master branch when reporting the issue

**Comment by jlewi** [comment]
@kunmingg @kkasravi has this been fixed?

**Comment by jlewi** [comment]
I think there's still a bug in the CRD  spec.

https://github.com/kubeflow/kubeflow/blob/fee939dcfc4530629be93e54bebba226e0dfbdbb/kubeflow/tf-training/tf-job-operator.libsonnet#L52

The scope in the CRD definition should always be namespace. TFJobs are always namespace scoped; meaning TFJob resources belong to a namespace.

The scope refers to parameters for the controller; depending on the parameters the controller will either handle resources in a specific namespace or all namespaces.

/cc @lluunn

**Comment by lluunn** [comment]
This is fixed in #1987
